### PR TITLE
add reason to getLockCategories

### PR DESCRIPTION
**Pull request by mchangrh**
blocked by #304

### CODE REVIEW

File: src/routes/getLockCategories.ts [hunk 1]
```diff
@@ -13,11 +13,15 @@ export async function getLockCategories(req: Request, res: Response): Promise<Re
...
-        const lockedCategories = await db.prepare('all', 'SELECT "category" from "lockCategories" where "videoID" = ?', [videoID]) as {category: Category}[];
-        if (lockedCategories.length === 0 || !lockedCategories[0]) return res.sendStatus(404);
-        // map to array in JS becaues of SQL incompatibilities
-        const categories = Object.values(lockedCategories).map((entry) => entry.category);
+        const row = await db.prepare('all', 'SELECT "category", "reason" from "lockCategories" where "videoID" = ?', [videoID]) as {category: Category, reason: string}[];
+        // map categories to array in JS becaues of SQL incompatibilities
+        const categories = row.map(item => item.category);
+        if (categories.length === 0 || !categories[0]) return res.sendStatus(404);
+        // Get longest lock reason
+        const reason = row.map(item => item.reason)
```
Comment (ajayyy) [inline src/routes/getLockCategories.ts:21]: Should we get the latest one instead?

Comment (mchangrh) [reply src/routes/getLockCategories.ts:21]: I assumed that the longest one would have the most information, while the latest might just be relying on an old warning message

Comment (ajayyy) [reply src/routes/getLockCategories.ts:21]: https://discord.com/channels/603643120093233162/603643299961503761/863870164108574731

Comment (mchangrh) [reply src/routes/getLockCategories.ts:21]: Got it, will use the latest reason

**Pull request merged**